lisa/tests/scheduler/load_tracking.py

```diff
@@ -121,11 +121,11 @@ def get_max_capa_cpu(cls, plat_info):
 
     @classmethod
     def get_task_duty_cycle_pct(cls, trace, task_name, cpu):
-        window = cls.get_task_window(trace, task_name, cpu)
 
-        top = Topology()
-        top.add_to_level('cpu', [[cpu]])
-        return SchedAssert(trace._ftrace, top, execname=task_name).getDutyCycle(window)
+        df = trace.analysis.tasks.df_task_total_residency(task_name)
+        run_time = df.iloc[cpu].values[0]
```

Comment:
Untested: wouldn't it be more robust with ?
```
df[cpu]['runtime']
```
`iloc` will take the n'th row, but according to the doc the CPU is the index. We could have:
```
CPU XXX
0       B
4       B
```
Which would give:
```
# Looking for CPU4 using `iloc` means accessing the 2nd row
df.iloc[1]['XXX'] == A
# This is the expected behavior, where the CPU ID is used as the index value, rather than as a row number.
df[4]['XXX'] == B
```
Columns (`runtime`) need to be accessed by names rather than making a Series from the row and then accessing the nth item. We make no guarantee on the order of columns, we might add new ones, and it makes the code much more readable.

Reply:
Right! I meant to use loc intead of iloc - I always get them mixed up. But I think you meant to say I should use: ``` run_time = df[cpu]['runtime'] ``` It id definitely a better solution. I'll fix it! Thanks!